otx/core/data/caching/storage_cache.py

```diff
@@ -0,0 +1,96 @@
+
+
+def arrow_cache_helper(
+    dataset: DatumDataset,
+    scheme: str,
+    num_workers: int = 0,
+    cache_dir: str = DATASET_CACHE,
+    force: bool = False,
+) -> List[str]:
+    """A helper for dumping Datumaro arrow format."""
+
+    def get_hash(source_path, scheme):
+        _hash = hashlib.sha256()
+        _hash.update(f"{source_path}".encode("utf-8"))
+        _hash.update(f"{scheme}".encode("utf-8"))
+        if source_path:
+            _hash.update(str(os.stat(source_path)[stat.ST_MTIME]).encode("utf-8"))
+            if os.path.isdir(source_path):
+                for root, dirs, files in os.walk(source_path):
+                    for file in files:
+                        file = os.path.join(root, file)
+                        _hash.update(str(os.stat(file)[stat.ST_MTIME]).encode("utf-8"))
+                    for _dir in dirs:
+                        _dir = os.path.join(root, _dir)
+                        _hash.update(str(os.stat(_dir)[stat.ST_MTIME]).encode("utf-8"))
+        return _hash.hexdigest()
+
+    def get_file_hash(file):
+        _hash = hashlib.sha256()
+        _hash.update(str(file).encode("utf-8"))
+        _hash.update(str(os.stat(file)[stat.ST_MTIME]).encode("utf-8"))
+        return _hash.hexdigest()
+
+    cache_dir = os.path.join(cache_dir, get_hash(dataset.data_path, scheme))
+    cache_paths = []
+    os.makedirs(cache_dir, exist_ok=True)
+
+    cache_hit = []
+    for cache_path in os.listdir(cache_dir):
+        if not cache_path.endswith(".arrow"):
+            continue
+        cache_hit.append(False)
+        cache_path = os.path.join(cache_dir, cache_path)
+        cache_paths.append(cache_path)
+        hash_path = f"{cache_path}.hash"
+        if os.path.exists(cache_path) and os.path.exists(hash_path) and not force:
+            with open(hash_path, "r", encoding="utf-8") as f:
+                if get_file_hash(cache_path) == f.read():
+                    cache_hit[-1] = True
+
+    if cache_hit and all(cache_hit):
+        return cache_paths
+
+    dataset.export(
+        cache_dir,
+        "arrow",
+        save_media=True,
+        image_ext=scheme,
+        num_workers=num_workers,
+    )
+
+    cache_paths = []
+    for cache_path in os.listdir(cache_dir):
+        if not cache_path.endswith(".arrow"):
+            continue
+        cache_path = os.path.join(cache_dir, cache_path)
+        cache_paths.append(cache_path)
+        hash_path = f"{cache_path}.hash"
+        with open(hash_path, "w", encoding="utf-8") as f:
+            f.write(get_file_hash(cache_path))
+
+    return cache_paths
+
+
+def init_arrow_cache(dataset: DatumDataset, scheme: Optional[str] = None, num_workers: int = 0) -> DatumDataset:
+    """Init arrow format cache from Datumaro."""
+    if scheme is None or scheme == "NONE":
+        return dataset
+    cache_paths = arrow_cache_helper(dataset, scheme, num_workers)
+    dataset = DatumDataset.import_from(os.path.dirname(cache_paths[0]), "arrow")
```

Comment:
With `os.path.dirname(cache_paths[0])`, is this going to load all data in `cache_paths`?

Reply:
`cache_paths` would contain all the arrow files exported. Pointing the parent directory of them would load all arrow files into datumaro dataset.